### PR TITLE
Ibc 2024

### PR DESCRIPTION
**Pull request by mejohnnaylor**
fixed constraints codegen, fixed setValue, added setValue / getValue methods to device model for connection managers to use to support the RPCs / API calls of the same name.

### CODE REVIEW

File: sdks/cpp/lite/include/ParamWithValue.h [hunk 1]
```diff
@@ -70,17 +72,40 @@ class ParamWithValue : public catena::common::IParam {
...
     }
 
+    /**
+     * @brief deserialize the parameter value from protobuf
+     * @param dst the destination value
+     * @param value the protobuf value to deserialize from
+     * @note should only be called for non structured types
+     */
+    void fromProto(void* dst, catena::Value& value) override {
+        assert(dst == &value_.get());
```
Comment (johndanenRV) [inline sdks/cpp/lite/include/ParamWithValue.h:94]: Why is this here? If dst and value.get() are the same then why not remove the dst parameter?

**Comment by mejohnnaylor** [comment]
overtaken by events
